packages/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy/lib/LinkQueueLogger.ts

```diff
@@ -0,0 +1,209 @@
+   * @param {IURLStatistic} statisticLink - current link
+   * @param {IReachabilityRatio} event - The current event
+   */
+  private appendReachabilityStatistic(statisticLink: IURLStatistic, event: keyof IReachabilityRatio): void {
+    if (statisticLink.reachability_criteria !== null &&
+      this.reachabilityRatio[event][statisticLink.reachability_criteria] !== undefined) {
+      this.reachabilityRatio[event][statisticLink.reachability_criteria] += 1;
+    } else if (statisticLink.reachability_criteria !== null &&
+      this.reachabilityRatio[event][statisticLink.reachability_criteria] === undefined) {
+      this.reachabilityRatio[event][statisticLink.reachability_criteria] = 1;
+    } else if (this.reachabilityRatio[event].unknown === undefined) {
+      this.reachabilityRatio[event].unknown = 1;
+    } else {
+      this.reachabilityRatio[event].unknown += 1;
+    }
+  }
+
+  public override push(link: ILink, parent: ILink): boolean {
+    const resp: boolean = super.push(link, parent);
+    if (resp) {
+      const reachabilityCriteriaParentLink = LinkQueueLogger.getReachability(parent);
+
+      const parentStatisticLink: IURLStatistic = {
+        url: parent.url,
+        reachability_criteria: reachabilityCriteriaParentLink === null ? null : reachabilityCriteriaParentLink[0],
+        reachability_criteria_dynamic_info: reachabilityCriteriaParentLink === null ?
+          undefined :
+          reachabilityCriteriaParentLink[1],
+      };
+
+      const reachabilityCriteriaCurrentLink = LinkQueueLogger.getReachability(link);
+
+      const statisticLink: IURLStatistic = {
+        url: link.url,
+        reachability_criteria: reachabilityCriteriaCurrentLink === null ? null : reachabilityCriteriaCurrentLink[0],
+        reachability_criteria_dynamic_info: reachabilityCriteriaCurrentLink === null ?
+          undefined :
+          reachabilityCriteriaCurrentLink[1],
+        timestamp: Date.now(),
+        parent: parentStatisticLink,
+      };
+      this.appendReachabilityStatistic(statisticLink, 'pushEvent');
+      const event: ILinkQueueEvent = {
+        type: EventType.Push,
+        link: statisticLink,
+        query: this.query,
+        queueStatistics: {
+          size: this.getSize(),
+          reachabilityRatio: this.reachabilityRatio,
+        },
+      };
+
+      this.materialize(event);
+    }
+    return resp;
+  }
+
+  public override pop(): ILink | undefined {
+    const link = super.pop();
+    if (link !== undefined) {
+      const reachabilityCriteriaCurrentLink = LinkQueueLogger.getReachability(link);
+      const statisticLink: IURLStatistic = {
+        url: link.url,
+        reachability_criteria: reachabilityCriteriaCurrentLink === null ? null : reachabilityCriteriaCurrentLink[0],
+        reachability_criteria_dynamic_info: reachabilityCriteriaCurrentLink === null ?
+          undefined :
+          reachabilityCriteriaCurrentLink[1],
+        timestamp: Date.now(),
```

Comment:
Would https://developer.mozilla.org/en-US/docs/Web/API/Performance/now be better?

Reply:
It seems like it ` Cannot assign to read only property` in nodejs 18, but it work with nodejs 20